modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v550/RegistryDataManager.java

```diff
@@ -0,0 +1,112 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.carbon.is.migration.service.v550;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.core.util.CryptoException;
+import org.wso2.carbon.core.util.CryptoUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.is.migration.internal.ISMigrationServiceDataHolder;
+import org.wso2.carbon.registry.api.Collection;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.registry.api.RegistryException;
+import org.wso2.carbon.registry.api.Resource;
+import org.wso2.carbon.user.api.Tenant;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import static org.wso2.carbon.base.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+import static org.wso2.carbon.context.RegistryType.SYSTEM_GOVERNANCE;
+
+public class RegistryDataManager {
+
+    private static final Log log = LogFactory.getLog(RegistryDataManager.class);
+
+    public static final String ENTITLEMENT_POLICY_PUBLISHER = "/repository/identity/entitlement/publisher/";
+    public static final String PASSWORD_PROPERTY = "subscriberPassword";
+
+    private static RegistryDataManager instance = new RegistryDataManager();
+
+    private RegistryDataManager(){}
+
+    public static RegistryDataManager getInstance() {
+        return instance;
+    }
+
+    private void startTenantFlow(Tenant tenant) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(tenant.getId());
+        carbonContext.setTenantDomain(tenant.getDomain());
+    }
+
+    public void migrateSubscriberPassword(boolean migrateActiveTenantsOnly) throws UserStoreException {
+
+        //migrating super tenant configurations
+        try {
+            migrateSubscriberDataForTenant();
+            log.info("Policy Subscribers migrated for tenant : " + SUPER_TENANT_DOMAIN_NAME);
+        } catch (Exception e) {
+            log.error("Error while migrating Policy Subscribers for tenant : " + SUPER_TENANT_DOMAIN_NAME, e);
+        }
+
+        //migrating tenant configurations
+        Tenant[] tenants = ISMigrationServiceDataHolder.getRealmService().getTenantManager().getAllTenants();
+        for (Tenant tenant : tenants) {
```

Comment:
possible NPE if tenants is null

Reply:
If there are no tenants, an empty array would be returned. Hence NPE is avoided